plugins/quickicon/eos/services/provider.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * @package                 Joomla.Plugin
+ * @subpackage              Quickicon.Eos
+ * @copyright               (C) 2023 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license                 GNU General Public License version 2 or later; see LICENSE.txt
+ * @phpcs                   :disable PSR1.Classes.ClassDeclaration.MissingNamespace
+ */
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Extension\PluginInterface;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Plugin\PluginHelper;
+use Joomla\Database\DatabaseInterface;
+use Joomla\DI\Container;
+use Joomla\DI\ServiceProviderInterface;
+use Joomla\Event\DispatcherInterface;
+use Joomla\Plugin\Quickicon\Eos\Extension\Eos;
+
+return new class () implements ServiceProviderInterface {
+    /**
+     * Registers the service provider with a DI container.
+     *
+     * @param   Container  $container  The DI container.
+     *
+     * @return  void
+     *
+     * @since __DEPLOY_VERSION__
+     */
+    public function register(Container $container)
+    {
+        $container->set(PluginInterface::class, function (Container $container) {
+            $dispatcher = $container->get(DispatcherInterface::class);
+            $plugin     = new Eos($dispatcher, Factory::getApplication()->getDocument(), (array)PluginHelper::getPlugin('quickicon', 'eos'));
+            $plugin->setApplication(Factory::getApplication());
+            $plugin->setDatabase($container->get(DatabaseInterface::class));
+
+            return $plugin;
+        });
```

Comment:
```suggestion
        $container->set(
            PluginInterface::class,
            function (Container $container) {
                $dispatcher = $container->get(DispatcherInterface::class);
                $plugin     = new Eos(
                    $dispatcher,
                    Factory::getApplication()->getDocument(),
                    (array) PluginHelper::getPlugin('quickicon', 'eos')
                );
                $plugin->setApplication(Factory::getApplication());
                $plugin->setDatabase($container->get(DatabaseInterface::class));

                return $plugin;
            }
        );
```
format like other plugins